admin/acceptance.py

```diff
@@ -1289,24 +1291,31 @@ def capture_upstart(reactor, host, output_file):
             (b"flocker", b"flocker-container-agent"),
             (b"flocker", b"flocker-docker-plugin"),
             (b"upstart", b"docker")]:
-        path = FilePath(b'/var/log/').child(directory).child(service + b'.log')
-        formatter = TailFormatter(output_file, host, service)
-        ran = run_ssh(
-            reactor=reactor,
-            host=host,
-            username='root',
-            command=[
-                b'tail',
-                b'-F',
-                path.path
-            ],
-            handle_stdout=formatter.handle_output_line,
+
+        def pull_logs_for_process(directory=directory, service=service):
+            path = FilePath(
+                b'/var/log/').child(directory).child(service + b'.log')
+            formatter = TailFormatter(output_file, host, service)
+            ran = run_ssh(
+                reactor=reactor,
+                host=host,
+                username='root',
+                command=[
+                    b'tail',
+                    b'-F',
+                    path.path
+                ],
+                handle_stdout=formatter.handle_output_line,
+            )
+            ran.addErrback(write_failure, logger=None)
+            # Deliver a final empty line to process the last message
+            ran.addCallback(lambda ignored, formatter=formatter:
+                            formatter.handle_output_line(b""))
+            return ran
+
```

Comment:
:question: Maybe add a comment explaining the reason for retrying the log collection.

I always thought it'd be even better to just download the complete logs once the acceptance tests have finished.

Or run ``flocker-diagnostics`` after the tests and download the diagnostics archive.

We'd just need to modify flocker-diagnostics to accept a start and end time to get logs from around the tests...or a journald marker on Centos.